source/material_model/rheology/composite_viscous_creep.cc

```diff
@@ -0,0 +1,238 @@
+            stress_iteration += 1;
+          }
+
+        // The creep stress is not the total stress, so we still need to do a little work to obtain the effective viscosity.
+        // First, we compute the stress running through the strain rate limiter
+        const double lim_stress = 2.*min_viscosities[composition]*(edot_ii - creep_stress/(2.*max_viscosities[composition]));
+
+        // Now we return the viscosity using the total stress
+        return (creep_stress + lim_stress)/(2.*edot_ii);
+      }
+
+
+
+      template <int dim>
+      std::pair<double, double>
+      CompositeViscousCreep<dim>::compute_strain_rate_and_derivative (const double creep_stress,
+                                                                      const double pressure,
+                                                                      const double temperature,
+                                                                      const DiffusionCreepParameters diffusion_creep_parameters,
+                                                                      const DislocationCreepParameters dislocation_creep_parameters,
+                                                                      const PeierlsCreepParameters peierls_creep_parameters,
+                                                                      const double min_viscosity,
+                                                                      const double max_viscosity) const
+      {
+        const std::pair<double, double> diff_edot_and_deriv = diffusion_creep.compute_strain_rate_and_derivative(creep_stress, pressure, temperature, diffusion_creep_parameters);
+        const std::pair<double, double> disl_edot_and_deriv = dislocation_creep.compute_strain_rate_and_derivative(creep_stress, pressure, temperature, dislocation_creep_parameters);
+        const std::pair<double, double> prls_edot_and_deriv = peierls_creep.compute_strain_rate_and_derivative(creep_stress, pressure, temperature, peierls_creep_parameters);
+
+        const double strain_rate = creep_stress/(2.*max_viscosity) + (max_viscosity/(max_viscosity - min_viscosity))*(diff_edot_and_deriv.first + disl_edot_and_deriv.first + prls_edot_and_deriv.first);
+        const double strain_rate_deriv = 1./(2.*max_viscosity) + (max_viscosity/(max_viscosity - min_viscosity))*(diff_edot_and_deriv.second + disl_edot_and_deriv.second + prls_edot_and_deriv.second);
+
+        return std::make_pair(strain_rate, strain_rate_deriv);
+      }
+
+
+
+      template <int dim>
+      void
+      CompositeViscousCreep<dim>::declare_parameters (ParameterHandler &prm)
+      {
+
+        prm.declare_entry ("Minimum strain rate", "1.4e-20", Patterns::Double(0.),
+                           "Stabilizes strain dependent viscosity. Units: \\si{\\per\\second}.");
+
+        // Viscosity iteration parameters
+        prm.declare_entry ("Strain rate residual tolerance", "1e-22", Patterns::Double(0.),
```

Comment:
Similar to above, I wonder if there is way we can declare these parameters once, and then use them for all of the relevant rheology modules?

Reply:
I don't like the idea of having implicit declarations in the material models. Redeclaring the same entry just overwrites the previous default, so there shouldn't be a problem.

I've changed the ordering of the declarations so that the default given here overrides whatever is given in the rheology modules.